### PR TITLE
Make deployments configurable

### PR DESCRIPTION
**Pull request by NeonDaniel**
# Description
Add options to disable non-essential backend services
Disable services for which configuration was skipped
Refactors `iris` core service to be optional
Adds image tag configuration to core deployment CLI command

# Issues
Fixes typo in Klat configuration from #56

# Other Notes

### CODE REVIEW

**Review by mikejgray — COMMENTED** [review]
On my plate to review today

File: neon_diana_utils/configuration.py [hunk 1]
```diff
@@ -593,23 +645,48 @@ def configure_neon_core(mq_user: str = None,
...
-                                   default=iris_domain)
-        confirmed = click.confirm(f"Is {iris_domain} correct?")
+    iris_domain = None
+    if click.confirm("Configure IRIS Gradio Web UI?"):
+        confirmed = False
+        iris_domain = "iris.diana.k8s"  # TODO: Read from backend config
+        while not confirmed:
+            iris_domain = click.prompt("Hostname for Iris Gradio Web UI",
+                                       type=str, default=iris_domain)
+            confirmed = click.confirm(f"Is {iris_domain} correct?")
```
Comment (mikejgray) [inline neon_diana_utils/configuration.py]: This logic doesn't quite work with confirmation - if you make a mistake, `iris_domain` becomes the previous mistaken value:

```sh
Hostname for Iris Gradio Web UI [iris.diana.k8s]: y
Is y correct? [y/N]: n
Hostname for Iris Gradio Web UI [y]: iris.diana.k8s
Is iris.diana.k8s correct? [y/N]: y
```

**Review by mikejgray — CHANGES_REQUESTED** [review]
I don't think this is quite generating properly. Tree output for an almost full installation:

```sh
diana@diana:~/neon_diana$ tree .
.
├── chatbots
│   ├── Chart.yaml
│   ├── chatbots.yaml
│   ├── templates
│   │   └── config-map.yaml
│   └── values.yaml
├── diana-backend
│   ├── Chart.yaml
│   ├── diana.yaml
│   ├── rabbitmq.json
│   ├── templates
│   │   └── config-secret.yaml
│   └── values.yaml
└── neon-core
    ├── Chart.yaml
    ├── neon.yaml
    ├── templates
    │   └── config-map.yaml
    └── values.yaml

7 directories, 13 files
```

And for a minimal installation:

```sh
diana@diana:~/minimal_diana$ tree .
.
├── diana-backend
│   ├── Chart.yaml
│   ├── diana.yaml
│   ├── rabbitmq.json
│   ├── templates
│   │   └── config-secret.yaml
│   └── values.yaml
└── neon-core
    ├── Chart.yaml
    ├── neon.yaml
    ├── templates
    │   └── config-map.yaml
    └── values.yaml

4 directories, 9 files
```

It looks like we're missing a lot of subdirectories and even some whole services.

**Comment by NeonDaniel** [comment]
> It looks like we're missing a lot of subdirectories and even some whole services.

Did you `helm dependency build diana-backend`, etc first?

File: neon_diana_utils/configuration.py (hunk 1 above)
Comment (NeonDaniel) [reply neon_diana_utils/configuration.py]: Would it make more sense to set the default to `None` if not confirmed? I didn't really consider the default value behavior after an unconfirmed entry

Comment (mikejgray) [reply neon_diana_utils/configuration.py]: Probably easiest to just set a different variable from `iris_domain` on line 653, right? Reusing the variable is what's causing the issue, I think

Comment (NeonDaniel) [reply neon_diana_utils/configuration.py]: I guess if not confirmed then there's no reason to use the invalid thing as a default.. ~I'll just hard-code the default in the prompt~ I see the TODO to read from config so I'll look into doing that

**Review by mikejgray — APPROVED** [review]
This appears to be generating properly now, with the understanding that a lot of it comes up the Helm repository online rather than being generated locally

**Pull request merged**